### PR TITLE
App 944 update salesforce parser and renderer

### PR DESCRIPTION
**Pull request by crepache**
Created structure of Versions and migration of "Opportunity Notification" for V2.

### CODE REVIEW

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    formatOptionalField(node, SalesforceConstants.STAGE_NAME, stageName);
+  }
+
+  private void processProbability(JsonNode node) {
+    String probability = node.path(SalesforceConstants.PROBABILITY).asText(EMPTY);
+
+    formatOptionalField(node, SalesforceConstants.PROBABILITY, probability);
+  }
+
+  private void processUpdatedFields(JsonNode nodeCurrent, JsonNode nodePrevious) {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java]: currentNode, previousNode

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+    if (!StringUtils.isEmpty(updatedFields)) {
+      ((ObjectNode) nodeCurrent).put(SalesforceConstants.UPDATED_FIELDS, updatedFields);
+      ((ObjectNode) nodeCurrent).put(SalesforceConstants.CREATED_OR_UPDATED, SalesforceConstants.UPDATED_NOTIFICATION);
+    } else {
+      ((ObjectNode) nodeCurrent).put(SalesforceConstants.CREATED_OR_UPDATED, SalesforceConstants.CREATED_NOTIFICATION);
+    }
+  }
+
+  private void formatOptionalField(JsonNode node, String nameNode, String valueNode) {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java]: nameName, nodeValue

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceFactory.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.ws.rs.core.MediaType;
+import javax.xml.bind.JAXBException;
+
+/**
+ * Created by crepache on 19/04/17.
+ */
+public abstract class SalesforceFactory implements WebHookParserFactory {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceFactory.java]: SalesforceParserFactory

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceResolver.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Resolves the parser factory should be used based on MessageML version supported by Agent.
+ *
+ * Created by crepache on 19/04/17.
+ */
+@Component
+public class SalesforceResolver extends WebHookParserResolver {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceResolver.java]: SalesforceParserResolver

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v1/NullSalesforceParser.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by crepache on 26/04/17.
+ */
+@Component
+public class NullSalesforceParser implements SalesforceParser {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v1/NullSalesforceParser.java:20]: Null Parser should return null, not an actual message.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+    return null;
+  }
+
+  @Override
+  public Message parse(Map<String, String> parameters, JsonNode node) throws SalesforceParseException {
+    return parse(node);
+  }
+
+  @Override
+  protected void preProcessInputData(JsonNode node) {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java]: This should be part of the OpportunityNotificationParser.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/OpportunityNotificationJSONMetadataParser.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by crepache on 19/04/17.
+ */
+@Component
+public class OpportunityNotificationJSONMetadataParser extends SalesforceMetadataParser {
```
Comment (ecarrenho-daitan) [inline src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/OpportunityNotificationJSONMetadataParser.java]: Change name to OpportunityNotificationMetadataParser

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java (hunk 1 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java]: Done.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java (hunk 2 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java]: Done.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceFactory.java (hunk 3 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceFactory.java]: Done.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceResolver.java (hunk 4 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/SalesforceResolver.java]: Done.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/OpportunityNotificationJSONMetadataParser.java (hunk 7 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/OpportunityNotificationJSONMetadataParser.java]: Done.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java (hunk 6 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v2/SalesforceMetadataParser.java]: Done.

File: src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v1/NullSalesforceParser.java (hunk 5 above)
Comment (crepache) [reply src/main/java/org/symphonyoss/integration/webhook/salesforce/parser/v1/NullSalesforceParser.java:20]: Done.

**Review by ecarrenho-daitan — APPROVED** [review]
LGTM

**Pull request merged**